contracts/modules/common/Utils.sol

```diff
@@ -42,7 +42,10 @@ library Utils {
             v := and(mload(add(_signatures, add(0x41,mul(0x41,_index)))), 0xff)
         }
         require(v == 27 || v == 28);
-        return ecrecover(_signedHash, v, r, s);
+
+        address recoveredAddress = ecrecover(_signedHash, v, r, s);
+        require(recoveredAddress != address(0), "Utils: ecrecover returned 0");
```

Comment:
We should also include the check agains signature maleability.

Reply:
I didn't include it as it's not in the recommendation due to the fact we don't allow signature re-use.